File: src/main/java/org/myfintech/payment/validator/PaymentValidator.java
/*
 * Copyright (c) 2024 MyFintech Payment Service
 * All rights reserved.
 *
 * This software is proprietary and confidential. Unauthorized copying of this file,
 * via any medium, is strictly prohibited.
 * @author : Dhanuka Ranasinghe
 * @since : Date: 11/07/2025
 */
package org.myfintech.payment.validator;

import jakarta.validation.ConstraintViolation;
import jakarta.validation.ConstraintViolationException;
import jakarta.validation.Validator;
import org.myfintech.payment.domain.PaymentDTO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import lombok.extern.slf4j.Slf4j;

import java.util.Set;

@Component
@Slf4j
public class PaymentValidator {

    private final Validator validator;

    @Autowired
    public PaymentValidator(Validator validator) {
        this.validator = validator;
    }

    public void validatePaymentRequest(PaymentDTO dto) {
        Set<ConstraintViolation<PaymentDTO>> violations = validator.validate(dto);
        if (!violations.isEmpty()) {
            log.warn("PaymentDTO validation failed: {}", violations);
            throw new ConstraintViolationException("Validation failed for PaymentDTO", violations);
        }
    }
}

File: src/main/kotlin/org/myfintech/payment/validator/PaymentValidator.kt
/*
 * Copyright (c) 2024 MyFintech Payment Service
 * All rights reserved.
 *
 * This software is proprietary and confidential. Unauthorized copying of this file,
 * via any medium, is strictly prohibited.
 * @author : Dhanuka Ranasinghe
 * @since : Date: 11/07/2025
 */
package org.myfintech.payment.validator

import jakarta.validation.ConstraintViolationException
import jakarta.validation.Validator
import org.myfintech.payment.domain.PaymentDTO
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class PaymentValidator(private val validator: Validator) {

    companion object {
        private val log: Logger = LoggerFactory.getLogger(PaymentValidator::class.java)
    }

    fun validatePaymentRequest(dto: PaymentDTO) {
        val violations = validator.validate(dto)
        if (violations.isNotEmpty()) {
            log.warn("PaymentDTO validation failed: {}", violations)
            throw ConstraintViolationException("Validation failed for PaymentDTO", violations)
        }
    }
}